core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -215,6 +215,12 @@ public void initialize(String name, Map<String, String> unresolved) {
     this.paths = ResourcePaths.forCatalogProperties(mergedProps);
 
     String token = mergedProps.get(OAuth2Properties.TOKEN);
+    // re-resolve these variables in case they were overridden by the config endpoint
+    credential = mergedProps.get(OAuth2Properties.CREDENTIAL);
+    scope = mergedProps.getOrDefault(OAuth2Properties.SCOPE, OAuth2Properties.CATALOG_SCOPE);
+    oauth2ServerUri =
+        mergedProps.getOrDefault(OAuth2Properties.OAUTH2_SERVER_URI, ResourcePaths.tokens());
```

Comment:
It's a good idea to get certain configs from server. Thanks for doing this. Should we add these into config endpoint spec if we are doing this? Without something in the spec, we are relying on the convention. There is a minor concern that the config response from the server side may happen to have the same name with different meaning. For example, a property named `scope` in the config response may mean different things. cc @danielcweeks

Reply:
Hi @flyrain thank you so much for reviewing!

Regarding the risk of config name clashes, most properties have a prefix such as `s3.`, `adls.`, `http-client.`, `ecs.`, etc. This prefix makes name clashes rather unlikely, while also clearly indicating the "scope" the property belongs to.

Unfortunately, none of the properties in `OAuth2Properties` have such a prefix, which makes them rather prone to misinterpretation indeed. 

This is imho a bit out of the scope if this PR, but I agree with you that we could probably do something about it, e.g. introduce a prefix `auth.` that the client would then automatically "map" to properties in `OAuth2Properties`. But we'd still need to support "unprefixed" properties for a long time I'm afraid.

Reply:
It's not a blocker for me. The current config response is a flat key-value pairs. I think it works well since they are only for client side catalog configs. We can bring more structures into to it in the future, as you suggested, e.g., a prefix `auth.`

Reply:
I don't think allowing override of the the auth server uri is a good idea.  The default should either be the spec'ed auth server path or something that is controlled by the client.  This would allow the client to be redirected to a server that they would not expect or be in control of.

Reply:
> The default should (...) be the spec'ed auth server path

I strongly disagree with this, the `/v1/oauth/tokens` should, as I mentioned above, have really never become part of the Iceberg REST spec:
* Each client passes the just base64 encoded credentials to that endpoint - that proxies requests to something you don't know - and gets some token back. This means, the resource knows your credentials.
* It is the opposite of what OAuth addresses: `OAuth addresses these issues by introducing an authorization layer and separating the role of the client from that of the resource owner.` - see [RFC 6749](https://datatracker.ietf.org/doc/html/rfc6749#section-1).